### ISSUE TITLE
dry-rb: uninitialized constant Concurrent::Array (NameError)

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1438]
When I tried on of my project I noticed one interesting issue with `truffleruby-1.0.0.rc8`. When one of dry-rb libraries is loaded using Bundler, `uninitialized constant Concurrent::Array (NameError)` is issued.

Here is a small reproducible example:

```ruby
require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'

  gem 'dry-struct', '0.6.0'
end
```

When I require`dry-struct` in `irb`, no issue is occured.

**Comment by pitr-ch** [comment]
@deepj please use concurrent-ruby 1.1.0.pre2 it's fixed there.

**Comment by deepj** [comment]
@pitr-ch This doesn't fail on MRI. And unfortuntaly, `dry-configurable` on which `dry-struct` depends, supports recently only `concurrent-ruby` 1.0.x :(

**Comment by pitr-ch** [comment]
Yeah it works in MRI. `Concurrent::Array` has different implementations for each Ruby implementation. The implementation for TruffleRuby broke in 1.0.x as we continued developing TruffleRuby and we did not caught it.

`dry-configurable` depends with `concurrent-ruby` ~> '1.0' therefore when `gem 'concurrent-ruby', '>= 1.1.0.pre2'` is added into the Gemfile it should work.

**Comment by deepj** [comment]
@pitr-ch sorry, I overlooked the condition in `dry-configurable` requiring `concurrent-ruby`. It _works_ now. Then, this issue can be closed.

**Comment by pitr-ch** [comment]
Great! I will release concurrent-ruby 1.1 soon to remove the need for this.
